### PR TITLE
[sparkle] - feat(DataTable): implement pagination

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at adding the ability to paginate within a `DataTable`.

## Risk

Low

## Deploy Plan

Deploy `sparkle`

### CODE REVIEW

File: sparkle/src/components/DataTable.tsx [hunk 1]
```diff
@@ -11,13 +11,16 @@ import {
...
-  DropdownItemProps,
+  Avatar,
   DropdownMenu,
-} from "@sparkle/components/DropdownMenu";
-import { IconButton } from "@sparkle/components/IconButton";
-import { ArrowDownIcon, ArrowUpIcon, MoreIcon } from "@sparkle/icons";
+  IconButton,
+  MoreIcon,
+  Pagination,
+} from "@sparkle/index";
```
Comment (tdraier) [inline sparkle/src/components/DataTable.tsx:22]: please avoid circular imports, use imports from `@sparkle/components/xxx`

File: sparkle/src/components/DataTable.tsx [hunk 2]
```diff
@@ -11,13 +11,16 @@ import {
...
+import { ArrowDownIcon, ArrowUpIcon } from "@sparkle/index";
```
Comment (tdraier) [inline sparkle/src/components/DataTable.tsx:23]:
```suggestion
import { ArrowDownIcon, ArrowUpIcon } from "@sparkle/icons";
```

File: sparkle/src/components/DataTable.tsx [hunk 3]
```diff
@@ -89,81 +94,90 @@ export function DataTable<TData extends TBaseData>({
...
+            </DataTable.Row>
+          ))}
+        </DataTable.Body>
+      </DataTable.Root>
+      {paginationProps && (
+        <Pagination
+          itemsCount={paginationProps.itemsCount}
+          maxItemsPerPage={paginationProps.maxItemsPerPage}
+          onButtonClick={paginationProps.onButtonClick}
+        />
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:178]: `nit`
```suggestion
        <Pagination {...paginationProps} />
```

File: sparkle/src/components/DataTable.tsx [hunk 4]
```diff
@@ -89,81 +106,90 @@ export function DataTable<TData extends TBaseData>({
...
-                        }
-                        size="xs"
-                        className={classNames(
-                          "s-ml-1",
-                          header.column.getIsSorted()
-                            ? "s-opacity-100"
-                            : "s-opacity-0"
-                        )}
-                      />
+    <div className="s-mb-2 s-ml-2 s-flex s-flex-col s-gap-2">
```
Comment (flvndvd) [inline sparkle/src/components/DataTable.tsx:109]: Can we drop margin styles from here?

File: sparkle/src/components/PaginatedCitationsGrid.tsx [hunk 5]
```diff
@@ -65,12 +65,13 @@ export function PaginatedCitationsGrid({
...
-          itemsCount={items.length}
-          maxItemsPerPage={maxItemsPerPage}
+          rowCount={items.length}
+          pageSize={maxItemsPerPage}
           size="xs"
           showDetails={false}
           showPageButtons={false}
-          onButtonClick={(pageNb) => setCurrentPage(pageNb)}
+          pageIndex={currentPage}
+          setPageIndex={(pageNb) => setCurrentPage(pageNb)}
```
Comment (flvndvd) [inline sparkle/src/components/PaginatedCitationsGrid.tsx:74]:
```suggestion
          setPageIndex={setCurrentPage}
```

File: sparkle/src/components/Pagination.tsx [hunk 6]
```diff
@@ -10,43 +9,48 @@ type Size = "sm" | "xs";
...
 
 export function Pagination({
-  itemsCount,
-  maxItemsPerPage,
-  onButtonClick,
   size = "sm",
   showDetails = true,
   showPageButtons = true,
+  rowCount,
+  pageSize = 25,
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:27]: No magic variable can you store in a constant, please?

File: sparkle/src/components/Pagination.tsx [hunk 7]
```diff
@@ -10,43 +9,48 @@ type Size = "sm" | "xs";
...
+  pageIndex,
+  setPageIndex,
 }: PaginationProps) {
-  const numPages = Math.ceil(itemsCount / maxItemsPerPage);
+  const numPages = Math.ceil(rowCount / pageSize);
```
Comment (flvndvd) [inline sparkle/src/components/Pagination.tsx:31]: Small comment to explain that pageIndex is 0 index based?

**Review by flvndvd — APPROVED** [review]

**Review by flvndvd — APPROVED** [review]
LGTM once the pagination padding is fixed :)

**Pull request merged**